### PR TITLE
Validate availabilities directory

### PR DESCRIPTION
**Pull request by jtrees**
This PR...

- ... changes the way defaults and normalization are applied in the config.
- ... validates that the configured availabilities directory actually exists.

The Config class is getting a bit bloated. It might be a good idea to extract the filesystem relevant functionality into a class of its own at some point.

### CODE REVIEW

File: src/Grandeljay/Availability/Config.php [hunk 1]
```diff
@@ -54,20 +56,122 @@ private function loadConfig(): void
...
+    private function normalizeConfig(array $rawConfig): array
+    {
+        $normalizedConfig = $rawConfig; // Create a copy.
+
+        $normalizedConfig['directoryAvailabilities'] = $this->normalizeAvailabilitiesDir($rawConfig['directoryAvailabilities']);
+        $normalizedConfig['maxAvailabilitiesPerUser'] = $rawConfig['maxAvailabilitiesPerUser'] ?? 100;
+        $normalizedConfig['defaultDay'] = $rawConfig['defaultDay'] ?? "monday";
+        $normalizedConfig['defaultTime'] = $rawConfig['defaultTime'] ?? "19:00";
+        $normalizedConfig['eventName'] = $rawConfig['eventName'] ?? "Dota 2";
+        $normalizedConfig['logLevel'] = $rawConfig['logLevel'] ?? "Info";
```
Comment (grandeljay) [inline src/Grandeljay/Availability/Config.php:78]: Can you please align the equal signs to match the code style and use British English (`normalised`) instead?

File: src/Grandeljay/Availability/Config.php [hunk 2]
```diff
@@ -36,16 +36,18 @@ private function loadConfig(): void
             $potentialConfigPath = $this->getPathWithEnvironmentVariable($potentialConfigPath);
 
             if (file_exists($potentialConfigPath)) {
-                $raw_data    = file_get_contents($potentialConfigPath);
-                $parsed_data = json_decode($raw_data, true, 2, JSON_THROW_ON_ERROR);
-
-                $error = $this->validateConfig($parsed_data);
+                $rawData       = file_get_contents($potentialConfigPath);
+                $parsedData    = json_decode($rawData, true, 2, JSON_THROW_ON_ERROR);
+                $normalizedCfg = $this->normalizeConfig($parsedData);
```
Comment (grandeljay) [inline src/Grandeljay/Availability/Config.php:41]: Please use BE (`normalise`).

File: src/Grandeljay/Availability/Config.php [hunk 3]
```diff
@@ -36,16 +36,18 @@ private function loadConfig(): void
...
+                $error         = $this->validateConfig($normalizedCfg, $parsedData);
```
Comment (grandeljay) [inline src/Grandeljay/Availability/Config.php:42]: Please use BE (`normalise`).

File: src/Grandeljay/Availability/Config.php [hunk 4]
```diff
@@ -36,16 +36,18 @@ private function loadConfig(): void
...
 
                 if ($error) {
-                    die(sprintf('Bad config.json at `%s`: %s' . PHP_EOL, $potentialConfigPath, $error));
+                    $msg = sprintf('Bad config.json at `%s`:' . PHP_EOL, $potentialConfigPath);
+                    $msg = $msg . "  Error:       " . $error . PHP_EOL;
+                    die($msg);
                 }
 
-                $this->config = $parsed_data;
+                $this->config = $normalizedCfg;
```
Comment (grandeljay) [inline src/Grandeljay/Availability/Config.php:50]: Please use BE (`normalise`).

**Review by grandeljay — CHANGES_REQUESTED** [review]
Thanks, this looks really good!

I consider the American Language a fork of the original, British English language. Can you please use that for naming and such?

File: src/Grandeljay/Availability/Config.php (hunk 1 above)
Comment (jtrees) [reply src/Grandeljay/Availability/Config.php]: Done

File: src/Grandeljay/Availability/Config.php [hunk 5]
```diff
@@ -54,20 +56,122 @@ private function loadConfig(): void
...
+        }
+
+        $cwd = getcwd();
+
+        if (!$cwd) {
+            die('Could not determine current working directory.');
+        }
+
+        $segments = array($cwd, $path);
+        $absolutePath = implode(DIRECTORY_SEPARATOR, $segments);
```
Comment (grandeljay) [inline src/Grandeljay/Availability/Config.php:119]: OK, this is the last nit nitpick then I'm ready to merge. Can you please align these equal sings too? Thanks for everything, looks good!

**Review by grandeljay — CHANGES_REQUESTED** [review]

File: src/Grandeljay/Availability/Config.php (hunk 5 above)
Comment (jtrees) [reply src/Grandeljay/Availability/Config.php]: Ok, done.

**Review by grandeljay — APPROVED** [review]
LGTM

**Pull request merged**